app/errors/service_error.rb

```diff
@@ -0,0 +1 @@
+class ServiceError < Exception; end
```

Comment:
custom exception should inherit from StandardError

Reply:
also, if you want to pass params to the initializer, it needs to be defined like this:
```
class MyError < StandardError
  attr_reader :thing
  def initialize(msg="My default message", thing="apple")
    @thing = thing
    super(msg)
  end
end

begin
  raise MyError.new("my message", "my thing")
rescue => e
  puts e.thing # "my thing"
end
```

https://www.honeybadger.io/blog/ruby-custom-exceptions/